src/store/modules/user/actions.ts

```diff
@@ -52,24 +52,63 @@ const actions: ActionTree<UserState, RootState> = {
   /**
    * Get User profile
    */
-  async getProfile ( { commit }) {
+  async getProfile ( { commit, dispatch }) {
     const resp = await UserService.getProfile()
     if (resp.status === 200) {
       const localTimeZone = moment.tz.guess();
       if (resp.data.userTimeZone !== localTimeZone) {
         emitter.emit('timeZoneDifferent', { profileTimeZone: resp.data.userTimeZone, localTimeZone});
       }
+      if(resp.data.facilities.length > 0) {
+        const payload = {
+          "inputFields": {
+            "facilityId": resp.data.facilities[0].facilityId
+          },
+          "fieldList": ["locationSeqId", "areaId", "aisleId", "sectionId", "levelId", "positionId"],
+          "entityName": "FacilityLocation",
+          "distinct": "Y",
+          "noConditionFind": "Y"
+        }
+        await dispatch('getFacilityLocations', payload).then((locations: any) => {
+          resp.data.facilityLocations = locations;
+        })
+      }
+
       commit(types.USER_INFO_UPDATED, resp.data);
+      commit(types.USER_CURRENT_FACILITY_LOCATION_UPDATED, resp.data.facilityLocations.length > 0 ? resp.data.facilityLocations[0] : {});
       commit(types.USER_CURRENT_FACILITY_UPDATED, resp.data.facilities.length > 0 ? resp.data.facilities[0] : {});
     }
   },
 
   /**
    * update current facility information
    */
-  async setFacility ({ commit }, payload) {
+  async setFacility ({ state, commit, dispatch }, payload) {
+    const user = state.current as any;
+    
+    const param = {
+      "inputFields": {
+        "facilityId": payload.facility.facilityId
+      },
+      "viewSize": 20,
+      "fieldList": ["locationSeqId", "areaId", "aisleId", "sectionId", "levelId", "positionId"],
+      "entityName": "FacilityLocation",
+      "distinct": "Y",
+      "noConditionFind": "Y"
+    }
+
+    await dispatch('getFacilityLocations', param).then((locations: any) => {
+      user.facilityLocations = locations
```

Comment:
I think we should not append the result in user object, and we can directly commit the data in the then part only.